sdks/python/apache_beam/dataframe/frames.py

```diff
@@ -4625,9 +4625,48 @@ def repeat(self, repeats):
       raise TypeError("str.repeat(repeats=) value must be an int or a "
                       f"DeferredSeries (encountered {type(repeats)}).")
 
-  get_dummies = frame_base.wont_implement_method(
-      pd.core.strings.StringMethods, 'get_dummies',
-      reason='non-deferred-columns')
+  @frame_base.with_docs_from(pd.core.strings.StringMethods)
+  @frame_base.args_to_kwargs(pd.core.strings.StringMethods)
+  def get_dummies(self, **kwargs):
+    """
+    Series must be categorical type. Either cast to ``category`` to
+    infer categories, or preferred, cast to ``CategoricalDtype``
+    to ensure correct categories.
```

Comment:
A couple of points here:
- pandas documentation seems to prefer "dtype" rather than "type".
- casting to "category" to infer categories won't work for us, because we need to know the specific categories at construction time in order to figure out the proxy. But at construction time we don't have access to the data to infer the types from. So really the only option is if the users specifies the categories with `CategoricalDtype`. (Also, once we have a few operations that work this way, we should add some general documentation about how to do this at https://s.apache.org/dataframe-non-deferred-columns, which gets linked in error messages, and then we could link it here too)

Reply:
Ack on 1. 

And for 2, when I was referring to "cast to `category`", I meant that one could feasibly be doing the following
```
>>> s = pd.Series(['a', 'b', 'a', 'c', 'a', np.nan]).astype('category')
>>> s
0      a
1      b
2      a
3      c
4      a
5    NaN
dtype: category
Categories (3, object): ['a', 'b', 'c']
>>> s.dtype
CategoricalDtype(categories=['a', 'b', 'c'], ordered=False)
```
resulting in the inferred categorical types. This obviously can be fragile, but nevertheless technically works, and results in the `dtype` to be `CategoricalDtype`. If we explicitly define with `CategoricalDtype`, then we may get:
```
>>> s = pd.Series(['a', 'b', 'a', 'c', 'a', np.nan], dtype=pd.CategoricalDtype(categories=['b', 'a']))
>>> s
0      a
1      b
2      a
3    NaN
4      a
5    NaN
dtype: category
Categories (2, object): ['b', 'a']
>>> s.dtype
CategoricalDtype(categories=['b', 'a'], ordered=False)
```
Note that in the second example, I didn't pass in the category `c`, which will result in the unknown category (`NaN`) in row 3.  The onus for getting these categories correct would be on the user.

I can definitely I can add a dedicated documentation section about this.

Reply:
Yes but that example only works if you have a concrete pandas `Series`, our users will be working with Beam `DeferredSeries` objects, which don't support inferring a `CategoricalDtype` with `astype('category')`.